File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
         long long int sum=0; int pos=0;
        for(int i=0;i<chalk.size();i++)
        {
            sum+=chalk[i];
        }
        k=k%sum; 
        for(int i=0;i<chalk.size();i++)
        {
            k=k-chalk[i];
            if(k<0) {return i;} 
        }
        return 0;
    }
};

File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        long tot =0; 
        for(int i=0;i<chalk.length;i++)
        {
            tot+=chalk[i]; 
        }
        k%=tot;
        for(int i=0;i<chalk.length;i++)
        {
            k=k-chalk[i];
            if(k<0)
            {
                return i;
            }
        }
        return -1;
        
    }
}